tests/functional/adapter/test_python_model.py

```diff
@@ -96,3 +116,16 @@ def test_import(self, project):
             f"PUT file://{project.project_root}/seeds/iris.csv @dbt_integration_test/;"
         )
         run_dbt(["run"])
+
+
+# https://github.com/dbt-labs/dbt-snowflake/issues/393 is notorious for being triggered on some
+# environments but not others. As of writing this, we don't know the true root cause. This test may
+# not fail on all systems with problems regarding custom schema model configurations.
+class TestCustomSchemaWorks:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"custom_target_model.py": models__custom_target_model}
+
+    def test_custom_target(self, project):
+        results = run_dbt()
+        assert results[0].node.schema == f"{project.test_schema}_MY_CUSTOM_SCHEMA"
```

Comment:
On our system this is trivially true but won't be for all users. It's worth having this here if nothing else as one more watchdog for similar issues we encounter in the future.